src/main/java/eu/mcdb/ban_announcer/PunishmentAction.java

```diff
@@ -20,6 +20,11 @@
 import eu.mcdb.util.chat.ChatColor;
 import lombok.Data;
 import lombok.ToString;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.Statistic;

```

Comment:
It seems that these imports are not used, and also, you should not import bukkit classes here as this plugin supports BungeeCord too, and this will break on bungee as these classes are not present on in.

Reply:
I was trying some different things out to get the player's IP address and forgot some things obviously. Apologies

I'm not fluent with Java nor any other programming language, so I appreciate your tips. 

I've been teaching myself Java by reading others plugins and making small changes to it to see what happens. In this case I added something that I felt other people may find useful, but I did not realize that it had already been implemented, nor that my code was inefficient.

I've wrote a few plugins of my own, but overall I'm learning to code as a hobby. I don't plan on spending money on classes or anything, or making money with an ability to code.

I will not take these comments lightly, thank you.

Reply:
Hello, no need to apoligize!
I have also learned to program by modifying code (making small changes) ~ but that was 12 years ago when I was a little curious girl with no internet and an awesome [OLPC XO-1 laptop](https://es.wikipedia.org/wiki/OLPC_XO-1).

I wish you good luck and a lot of fun in your programming journey! Have a nice day/night :D